models/Amity.py

```diff
@@ -0,0 +1,221 @@
+    Room names are stored in lists.
+    '''
+    rooms = {"Office": [], "LivingSpace": []}
+    people = {
+        "FELLOWS": [],
+        "STAFF": []
+    }
+    allocated_persons = []
+    unallocated_persons = []
+
+    def create_room(self, room_name, room_type):
+        if type(room_name) != str:
+            print "Room name should be a string"
+            try:
+                sys.exit()
+            except:
+                print "Exception accomodates tdd"
+        if room_name in self.get_roomname(self.rooms["LivingSpace"]):
+            print "Cannot create duplicate rooms"
+            try:
+                sys.exit()
+            except:
+                print "Exception accomodates tdd"
+        if room_name in self.get_roomname(self.rooms["Office"]):
+            print "Cannot create duplicate rooms"
+            try:
+                sys.exit()
+            except:
+                print "Exception accomodates tdd"
+        if room_type == "LivingSpace":
+            livingspace = LivingSpace(room_name)
+            self.rooms["LivingSpace"].append(livingspace)
+            print "Living space {0} successfuly created".format(
+                livingspace.room_name)
+        if room_type == "Office":
+            office = Office(room_name)
+            self.rooms["Office"].append(office)
+            print "Room {0} successfuly created".format(
+                office.room_name)
+
+    def add_people(self, person_name, person_type, wants_space="N"):
+        if person_type == "FELLOW" and wants_space == "Y":
+            if type(person_name) == str:
+                try:
+                    if len(self.get_listofrooms()) != 0:
+                        fellow = Fellow(person_name)
+                        self.people["FELLOWS"].append(fellow)
+                        allocated = random.choice(self.get_listofrooms())
+                        fellow.allocated = True
+                        allocated.occupants.append(fellow)
+                        self.allocated_persons.append([fellow, allocated])
+                    else:
+                        fellow = Fellow(person_name)
+                        # fellow.allocated = False
+                        self.unallocated_persons.append(fellow)
+                        return "No rooms to add people to please create a room"
+                except IndexError:
+                    self.unallocated_persons.append(person_name)
+                    return "Extra person not allocated added to unallocated"
+        if person_type == "STAFF" and wants_space == "N":
+            staff = Staff(person_name)
+            self.people["STAFF"].append(staff)
+        elif person_type == "STAFF" and wants_space == "Y":
+            print "Staff not allowed to have living spaces"
+
+    def allocate_person(self, person_name, person_type, room_name):
+        person_obj = self.get_fellowobject(person_name)
+        if person_obj not in self.unallocated_persons:
```

Comment:
You're only allocating a person if they are not in the unallocated_persons list? Doesn't that mean that they are already allocated??